homeassistant/components/spotify/media_player.py

```diff
@@ -171,41 +142,39 @@ def volume_level(self) -> float | None:
         """Return the device volume."""
         if not self.currently_playing:
             return None
-        return self.currently_playing.get("device", {}).get("volume_percent", 0) / 100
+        return self.currently_playing.device.volume_percent / 100
 
     @property
     def media_content_id(self) -> str | None:
         """Return the media URL."""
-        if not self.currently_playing:
+        if not self.currently_playing or not self.currently_playing.item:
```

Comment:
make a decorator which returns `None` if item is None and passes item to the function if it isn't, something like
```py
@property
@ensure_item
def media_content_id(self, item: blah) -> str | None:
    """Return the media URL."""
   return item.uri
```